### PR TITLE
Align conda envs and add Windows 3.9 env

### PR DESCRIPTION
**Pull request by hectormz**
This PR reorders the conda environment files across operating systems to make them easier to maintain (and separates core and dev dependencies). It also adds Windows Python 3.9 env file.

### CODE REVIEW

File: conda-envs/environment-test-py39.yml [hunk 1]
```diff
@@ -12,16 +13,17 @@ dependencies:
...
 - pandas>=0.24.0
-- pre-commit>=2.8.0
-- pytest-cov>=2.5
-- pytest>=3.0
-- python-graphviz
 - python=3.9
+- python-graphviz
 - scipy>=1.4.1
 - typing-extensions>=3.7.4
+# Extra stuff for testing
```
Comment (canyon289) [inline conda-envs/environment-test-py39.yml]:
```suggestion
# Extra dependencies for testing
```

File: conda-envs/windows-environment-dev-py39.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# "dev" conda envs are to be used by devs in setting their local environments
+name: pymc-dev-py39
+channels:
+- conda-forge
+- defaults
+dependencies:
+ # base dependencies (see install guide for Windows)
```
Comment (canyon289) [inline conda-envs/windows-environment-dev-py39.yml]:
```suggestion
 # Base dependencies (see install guide for Windows)
```

File: conda-envs/windows-environment-dev-py39.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+- fastprogress>=0.2.0
+- h5py>=2.7
+- numpy>=1.15.0
+- pandas>=0.24.0
+- pip
+- python=3.9
+- python-graphviz
+- scipy>=1.4.1
+- typing-extensions>=3.7.4
+# Extra stuff for dev, testing and docs build
```
Comment (canyon289) [inline conda-envs/windows-environment-dev-py39.yml]:
```suggestion
# Extra dependencies for dev, testing and docs build
```

File: conda-envs/windows-environment-test-py39.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# "test" conda envs are used to set up our CI environment in GitHub actions
+name: pymc-test-py39
+channels:
+- conda-forge
+- defaults
+dependencies:
+ # base dependencies (see install guide for Windows)
```
Comment (canyon289) [inline conda-envs/windows-environment-test-py39.yml]:
```suggestion
 # Base dependencies (see install guide for Windows)
```

File: conda-envs/windows-environment-test-py39.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+- mkl-service==2.3.0
+- m2w64-toolchain
+- numpy>=1.15.0
+- pandas>=0.24.0
+- pip
+- python=3.9
+- python-graphviz
+- scipy>=1.4.1
+- typing-extensions>=3.7.4
+# Extra stuff for testing
```
Comment (canyon289) [inline conda-envs/windows-environment-test-py39.yml]:
```suggestion
# Extra dependencies for testing
```

**Comment by canyon289** [comment]
Thanks @hectormz!

**Comment by michaelosthege** [comment]
I like that the environment files with the comments are easier to understand, but we have a `pre-commit` check that automatically sorts them.
@hectormz could you check how that `pre-commit` hook works and if it could maybe sort only within each comment section?

If that's too complicated (very likely) we should discuss if we want to remove the automated sorting.

Personally, I'd be okay with that, since on all of my (Windows) machines that `pre-commit` hook always and annoyingly creates an empty line in the Windows env files that mustn't be committed.

**Comment by hectormz** [comment]
@michaelosthege Ah, forgot to install pre-commit in my new environment. The hook in its current form isn't sorting all the .yml in the folder anyway. I'll look into it. I also checked out `sort-simple-yaml` hook, but it's not simple enough

**Comment by michaelosthege** [comment]
@hectormz I asked folks on Slack and we'd be ok with removing the env sorting step from the pre-commit in favor of the explanatory comments you made here.

So feel free to do that commit and just let me know when the pre-commit passes locally, then I'll approve running the tests. (someone has to click this button every time you push..)

If you have other things on the schedule just let us know and someone can take over 🙂

**Comment by hectormz** [comment]
Thanks for the update @michaelosthege , just pushed! I also updated the script that generates pip requirements from conda env because it makes extra empty lines on Windows due to line ending.

**Comment by hectormz** [comment]
Did you want to add `windows-environment-dev-py39.yml` to the `tests.yml` GitHub action?

**Comment by michaelosthege** [comment]
> Did you want to add `windows-environment-dev-py39.yml` to the `tests.yml` GitHub action?

You mean to run the tests in that environment? In principle yes, but we'll want to make sure that we're not running any tests twice.
Also, there's another open PR right now that's about changing which Python versions the CI runs with. Please check to avoid git conflicts or opposing momentum

**Comment by hectormz** [comment]
Hi @michaelosthege  no need to make those changes here. I was just wondering out loud. I'm content to leave this PR focused on the changes made.

**Review by michaelosthege — APPROVED** [review]
LGTM, thanks!

@canyon289 can you give it a final look and hit the merge button if you're happy?

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
We will hopefully remove python version specific versions in #5911, but merging this already! Thanks for your help @hectormz

**Pull request merged**